content/realtime/authentication.textile

```diff
@@ -77,42 +77,110 @@ Here is a straightforward example of using Basic Authentication to connect to Ab
+
+Typically, in a browser environment, the <span lang="default">"@authUrl@":#auth-options</span><span lang="csharp">"@AuthUrl@":#auth-options</span> provided would be a relative URL to a local endpoint that is used to issue tokens to trusted clients. Client requests can, for example, be trusted based on session cookies.
+
+```[jsall]
+const realtime = new Ably.Realtime({ authUrl: '/auth' });
 ```
 
 ```[ruby]
-  realtime = Ably::Realtime.new(auth_url: 'http://my.website/auth')
+realtime = Ably::Realtime.new(auth_url: '/auth')
 ```
 
 ```[java]
-  ClientOptions options = new ClientOptions();
-  options.authUrl = "http://my.website/auth";
-  AblyRealtime realtime = new AblyRealtime(options);
+ClientOptions options = new ClientOptions();
+options.authUrl = "/auth";
+AblyRealtime realtime = new AblyRealtime(options);
 ```
 
 ```[objc]
-  ARTClientOptions *options = [[ARTClientOptions alloc] init];
-  options.authUrl = [NSURL URLWithString:@"http://my.website/auth"];
-  ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
+ARTClientOptions *options = [[ARTClientOptions alloc] init];
+options.authUrl = [NSURL URLWithString:@"/auth"];
+ARTRealtime *realtime = [[ARTRealtime alloc] initWithOptions:options];
 ```
 
 ```[swift]
-  let options = ARTClientOptions()
-  options.authUrl = NSURL(string: "http://my.website/auth")
-  let realtime = ARTRealtime(options: options)
+let options = ARTClientOptions()
+options.authUrl = NSURL(string: "/auth")
+let realtime = ARTRealtime(options: options)
 ```
 
 ```[csharp]
-  ClientOptions options = new ClientOptions();
-  options.AuthUrl = new Uri("http://my.website/auth");
-  AblyRealtime realtime = new AblyRealtime(options);
+ClientOptions options = new ClientOptions();
+options.AuthUrl = new Uri("/auth");
+AblyRealtime realtime = new AblyRealtime(options);
+```
+
+h4(#auth-options). How to specify auth options
+
+There are several "auth options":/realtime/authentication#auth-options you can specify along with @authUrl@ and @authCallback@:
+
+* @authMethod@ - when @authUrl@ is called, the default "GET" method will be used, unless "POST" is specified.
+* @authHeaders@ - allows you to pass additional headers as required, depending on your use case.
+* @authParams@ - allows you to pass additional query parameters, depending on your use case.
+
+```[jsall]
+const ably = new Ably.Realtime({ authUrl: "/auth", authMethod: "POST", authParams: { p1: param1, b: param2}, authHeaders: {h1: header1, h2: header2} });
+```
+
+h3(#auth-callback). Token refresh using @authCallback@
+
+For non-browser clients, an authentication callback is the preferred method of implementing token refresh, although browser clients can also use @authCallback@ if the use case requires it. This relies on your application communicating securely with your own servers to obtain a token.
+
+The following code sample shows an example of using an @authCallback@ in a client:
+
+```[jsall](code-editor:realtime/auth-token-callback)
```

Comment:
This code block tag says 'jsall' but uses the NodeJS standard library so I'm wondering if it should say 'jsnode' or something similar?